code/lib/core-server/src/presets/common-preset.ts

```diff
@@ -349,3 +349,22 @@ export const experimental_serverChannel = async (
 
   return channel;
 };
+
+/**
+ * Try to resolve react and react-dom from the root node_modules of the project
+ * addon-docs uses this to alias react and react-dom to the project's version when possible
+ * If the user doesn't have an explicit dependency on react this will return the existing values
+ * Which will be the versions shipped with addon-docs
+ */
+export const resolvedReact = async (existing: any) => {
+  try {
+    return {
+      ...existing,
+      react: dirname(require.resolve('react/package.json')),
+      reactDom: dirname(require.resolve('react-dom/package.json')),
+      mdx: dirname(require.resolve('@mdx-js/react/package.json')),
+    };
+  } catch (e) {
+    return existing;
+  }
+};
```

Comment:
I don't understand why we need this (Also, it shouldn't work in pnp environments). So either we get the react, react-dom and mdx package resolved from `@storybook/react` or from `@storybook/addon-docs`. Why must we do the same lookup in `@storybook/core-server`?